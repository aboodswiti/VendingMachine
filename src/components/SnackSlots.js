import React from "react";

//  Display All Snacks and Selected snack by get the item as props from SnackMachine
//  item is Object, item = { location: , item: , price: ,quantity: ,  img:  }

const SnackSlots = ({ item }) => {
  return (
    <div data-testid={`SnackItem-${item.location}`} className="item">
      <img className="item__img" src={item.img} alt="item" />
      <div className="item__info">
        <h3>{item.item}</h3>
        <h3>${item.price}</h3>
      </div>
      <h3 className="item__quantity">#{item.quantity}</h3>
      <h3 className="item__location">{item.location}</h3>
    </div>
  );
};

export default SnackSlots;
