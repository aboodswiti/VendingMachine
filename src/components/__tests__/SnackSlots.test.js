import {render , screen, cleanup} from "@testing-library/react"
import SnackSlots from "../SnackSlots"
import renderer from "react-test-renderer"
afterEach(() => {
    cleanup();
});

test('Should render the Snack item', ()=> {
    const snack = {
        location: "11",
        item: "cola",
        price: 2.1,
        quantity: 5,
        img: "https://media.officedepot.com/image/upload/b_rgb:FFFFFF,c_pad,dpr_1.0,f_auto,h_1665,q_auto,w_1250/c_pad,h_1665,w_1250/v1/products/208206/208206_p?pgw=1&pgwact=1",
      }

    render(<SnackSlots item={snack}/>)
    const SnackElement = screen.getByTestId("SnackItem-11");
expect(SnackElement).toBeInTheDocument();
expect(SnackElement).toHaveTextContent("cola");
expect(SnackElement).toHaveTextContent(2.1);
expect(SnackElement).toContainHTML("<h3");


})
test('SnackSlots,Should Matches snapshot', ()=> {
    const snack = {
    location: "12",
    item: "Pepsi",
    price: 2.5,
    quantity: 5,
    img: "https://www.luluhypermarket.com/medias/424982-01.jpg-515Wx515H?context=bWFzdGVyfGltYWdlc3wxNzk0MDh8aW1hZ2UvanBlZ3xoYjcvaDZhLzk3NzU1NTIzMzE4MDYvNDI0OTgyLTAxLmpwZ181MTVXeDUxNUh8YzE5MDY0NTE5ODE3NWIyMjVmZjRmNGUyNTRjNTc5NTAzMDZjMTdmZjQwNjkxMjc5NzE0MjdiMDg3MTQwNjViYw",
  }
const tree = renderer.create(<SnackSlots item={snack}/> ).toJSON();
expect(tree).toMatchSnapshot();
} );