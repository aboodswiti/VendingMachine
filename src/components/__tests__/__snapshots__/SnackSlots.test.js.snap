// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackSlots,Should Matches snapshot 1`] = `
<div
  className="item"
  data-testid="SnackItem-12"
>
  <img
    alt="item"
    className="item__img"
    src="https://www.luluhypermarket.com/medias/424982-01.jpg-515Wx515H?context=bWFzdGVyfGltYWdlc3wxNzk0MDh8aW1hZ2UvanBlZ3xoYjcvaDZhLzk3NzU1NTIzMzE4MDYvNDI0OTgyLTAxLmpwZ181MTVXeDUxNUh8YzE5MDY0NTE5ODE3NWIyMjVmZjRmNGUyNTRjNTc5NTAzMDZjMTdmZjQwNjkxMjc5NzE0MjdiMDg3MTQwNjViYw"
  />
  <div
    className="item__info"
  >
    <h3>
      Pepsi
    </h3>
    <h3>
      $
      2.5
    </h3>
  </div>
  <h3
    className="item__location"
  >
    12
  </h3>
  <h3
    className="item__quantity"
  >
    #
    5
  </h3>
</div>
`;
